lib/services/awsAdapter.js

```diff
@@ -44,6 +43,28 @@ function AWSAdapter (ec2) {
 			return result;
 		});
 	});
+
+	this.getPublicIPAddress = Bluebird.method(function (instanceID) {
+		return this.describeInstance(instanceID)
+		.then(function (instance) {
+			return	Bluebird.resolve(instance[ 0 ].PublicIpAddress);
+		});
+	});
+	this.describeInstance = Bluebird.method(function (instanceID) {
+		var params = {
+			InstanceIds : [ instanceID ]
+		};
+		return ec2.describeInstancesAsync(params)
+		.then(function (data) {
+			var instanceInfo = [];
+			data.Reservations.forEach(function (reservation) {
+				reservation.Instances.forEach(function (instance) {
+					instanceInfo.push(instance);
+				});
+			});
+			return Bluebird.resolve(instanceInfo);
```

Comment:
The `Bluebird.resolve()` call here is unnecessary. If you simply return the value then Bluebird will use it as the promise resolution automatically.